File: storm-dialect-tck/src/main/java/st/orm/tck/model/Owner.java
package st.orm.tck.model;

import lombok.Builder;
import org.jspecify.annotations.Nullable;
import st.orm.Entity;
import st.orm.PK;
import st.orm.Version;

@Builder(toBuilder = true)
public record Owner(
        @PK Integer id,
        String firstName,
        String lastName,
        Address address,
        @Nullable String telephone,
        @Version int version
) implements Entity<Integer> {}

File: storm-kotlinx-serialization/src/test/kotlin/st/orm/serialization/model/Owner.kt
package st.orm.serialization.model

import st.orm.Entity
import st.orm.Json
import st.orm.PK

/**
 * Simple domain object representing an owner.
 */
data class Owner(
    @PK val id: Int = 0,
    override val firstName: String,
    override val lastName: String,
    @Json val address: Address,
    val telephone: String?
) : Entity<Int>, Person
